### PR TITLE
MMCA-2856: Refactored and fixed findings

### PR DESCRIPTION
**Pull request by DamienButterworth**

### CODE REVIEW

**Review by lhepto — APPROVED** [review]
LGTM

File: app/services/DateTimeService.scala [hunk 1]
```diff
@@ -22,6 +22,6 @@ import javax.inject.Singleton
 
 @Singleton
 case class DateTimeService() {
```
Comment (derekspalmer) [inline app/services/DateTimeService.scala:24]: case object?

File: app/services/MetricsReporterService.scala [hunk 2]
```diff
@@ -40,11 +40,11 @@ class MetricsReporterService @Inject()(metrics: com.kenshoo.play.metrics.Metrics
         case Failure(exception: UpstreamErrorResponse) => exception.statusCode
         case Failure(_) => Status.INTERNAL_SERVER_ERROR
       }
-      updateResponseTimeHistogram(resourceName, httpResponseCode, startTime, dateTimeService.getTimeStamp())
+      updateResponseTimeHistogram(resourceName, httpResponseCode, startTime, dateTimeService.getTimeStamp)
     }
   }
 
-  def updateResponseTimeHistogram(resourceName: String, httpResponseCode: Int,
+  private def updateResponseTimeHistogram(resourceName: String, httpResponseCode: Int,
```
Comment (derekspalmer) [inline app/services/MetricsReporterService.scala:47]: :+1:

**Review by derekspalmer — COMMENTED** [review]
LGTM

**Pull request merged**